app/controllers/students_controller.rb

```diff
@@ -18,8 +18,12 @@ def edit
   end
 
   def update
-    Student.update(params[:id], stud_params)
-    redirect_to students_path
+    @student = Student.update(params[:id], stud_params)
```

Comment:
An additional error case that we're not handling here is when `params[:id]` refers to a non-existent `Student`. In that case `update` will throw `ActiveRecord::RecordNotFound` so we should probably handle that as well.

Note to @Hamled: In a later change Kari's got some code that redirects in `show` when a `Student` with the supplied ID is not found. We could possibly move that into a controller filter run before both of these actions (and `edit` as well, probably).
If we want to go completely overboard (and who doesn't love going overboard) we could use the [decent_exposure gem](https://github.com/hashrocket/decent_exposure).